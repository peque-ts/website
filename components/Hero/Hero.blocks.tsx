import React from 'react';

interface AnchorProps {
  to: string;
}

export const Anchor: React.FC<AnchorProps> = ({ to, children }) => (
  <a
    className="text-lg text-secondary-500 transition duration-300 hover:text-white"
    href={`#${to}`}
  >
    {children}
  </a>
);
