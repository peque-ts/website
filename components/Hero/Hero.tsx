import Link from 'next/link';
import React from 'react';

import { ProjectId, PROJECTS } from '../../lib/data';
import { Icon } from '../Icon';

export const Hero: React.VFC = () => {
  const renderLink = (projectId: ProjectId) => (
    <Link href={`/docs/${projectId}`}>
      <a className="text-lg text-secondary-300 transition duration-300 hover:text-white">
        {PROJECTS[projectId].name}
      </a>
    </Link>
  );

  return (
    <section className="h-80 mt-16 bg-gradient-to-b from-secondary-600 to-transparent flex items-center justify-center">
      <div className="flex items-start space-x-2">
        <Icon alt="Logo" name="logo" size={40} />
        <div>
          <h1>Peque</h1>
          <div className="flex flex-col space-y-1">
            {renderLink('framework')}
            {renderLink('graphql')}
            {renderLink('di')}
            {renderLink('smb')}
          </div>
        </div>
      </div>
    </section>
  );
};
