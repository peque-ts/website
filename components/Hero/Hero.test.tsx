import { render, screen } from '@testing-library/react';

import { Hero } from './Hero';

describe('Hero', () => {
  it('should render', () => {
    const { asFragment } = render(<Hero />);

    expect(screen.getAllByRole('link')).toHaveLength(3);
    expect(asFragment()).toMatchSnapshot();
  });
});
