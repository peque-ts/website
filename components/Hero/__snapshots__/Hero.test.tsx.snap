// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should render 1`] = `
<DocumentFragment>
  <section
    class="h-80 bg-gradient-to-b from-primary-600 to-primary-700 flex items-center justify-center"
  >
    <div
      class="flex items-start space-x-2"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2746%27%20height=%2746%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Logo"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div>
        <h1>
          Peque
        </h1>
        <div
          class="flex flex-col space-y-1"
        >
          <a
            class="text-lg text-secondary-500 transition duration-300 hover:text-white"
            href="#framework"
          >
            Framework
          </a>
          <a
            class="text-lg text-secondary-500 transition duration-300 hover:text-white"
            href="#di"
          >
            IoC Container
          </a>
          <a
            class="text-lg text-secondary-500 transition duration-300 hover:text-white"
            href="#smb"
          >
            Message Broker
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
